### PR TITLE
Add default profile generation into operator deployment

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
We now generate the default profiles from the `profiles/` directory into
the operator deployment. This includes an additional verification step
in the CI, where we check if the profiles in the deployment are
up-to-date.
#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes-sigs/seccomp-operator/issues/42

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
- Add default operator profiles to the deployment. For now we added an nginx:1.19.1 profile
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
Working on a fix for:
```
The ConfigMap "default-profiles" is invalid: data[nginx:1.19.1.json]: Invalid value: "nginx:1.19.1.json": a valid config key must consist of alphanumeric characters, '-', '_' or '.' (e.g. 'key.name',  or 'KEY_NAME',  or 'key-name', regex used for validation is '[-._a-zA-Z0-9]+')
```

File: deploy/operator.yaml [hunk 1]
```diff
@@ -111,4 +111,15 @@ spec:
...
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: seccomp-operator
+  name: default-profiles
+  annotations:
+    seccomp.security.kubernetes.io/profile: "true"
+data:
+  nginx-1.19.1.json: |-
```
Comment (hasheddan) [inline deploy/operator.yaml:124]: I have mentioned this in a few PR's, but we should think about making some of the deployment (such as the install of default profiles) configurable 👍

File: Makefile [hunk 2]
```diff
@@ -100,6 +104,10 @@ $(BUILD_DIR)/verify_boilerplate.py: $(BUILD_DIR)
 verify-go-mod: go-mod ## Verify the go modules
 	hack/tree-status
 
+.PHONY: verify-default-profiles
+verify-default-profiles: default-profiles ## Verify the generated default profiles
```
Comment (hasheddan) [inline Makefile:108]: Would it make sense to clear the profiles before regenerating them? That would guard against the case where the profiles were manually updated to be correct, but the generator was actually failing to run or not changing anything

File: profiles/build.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+type matcher struct {
+	foundConfigMapType bool
+	foundConfigMapName bool
+	foundData          bool
+	replace            bool
+}
+
+func main() {
+	file, err := os.Open(deploymentYAML)
```
Comment (hasheddan) [inline profiles/build.go:47]: I might take a pass after this is merged and working to see if we can make this a more general purpose tool for converting any seccomp profiles to a `ConfigMap`

File: deploy/operator.yaml (hunk 1 above)
Comment (saschagrunert) [reply deploy/operator.yaml:124]: I'll follow-up on that. I have no clue how we wanna do that without adding another level of abstraction. Do we wanna make it configurable via environment variables, Helm or a config file? I think a config file could be a good first step, which could be deployed in parallel to the deploy/operator.yaml. WDYT?

File: profiles/build.go (hunk 3 above)
Comment (saschagrunert) [reply profiles/build.go:47]: Yeah that would be cool, thank you! :pray:

File: Makefile (hunk 2 above)
Comment (saschagrunert) [reply Makefile:108]: Hm, we indirectly clear them by regeneration all at once. If the generation outputs the same as the state of the repository, then we should be fine in any case. :shrug:

**Comment by saschagrunert** [comment]
Thanks for the review @hasheddan, let's iterate on the configuration and the tooling. :pray: 
/hold cancel

**Pull request merged**